packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -635,12 +647,26 @@ export abstract class BaseService extends Resource
       throw new Error('Deployment alarms requires the ECS deployment controller.');
     }
 
+    if (
+      props.deploymentController?.type === DeploymentControllerType.CODE_DEPLOY
+      && props.taskDefinitionRevision
+      && props.taskDefinitionRevision.revisionId !== 'latest'
+    ) {
+      throw new Error('CODE_DEPLOY deploymentController cannot be used with a non-latest task definition revision');
+    }
+
```

Comment:
```suggestion
```
I don't think it's necessary.
If `props.deploymentController?.type === DeploymentControllerType.CODE_DEPLOY` we will run the first `if` statement and `taskDefinitionRevision` will just get ignored.

Reply:
The reason I did this is because if the user _specifically asked_ for revision 23, but they got `latest` instead, that seems like an unintended surprise/footgun

Reply:
Thanks for clarifying, however, I think it's overkill to fail a deployment that would work.
Let's use `Annotations.addWarningV2` to provide some feedback and be more user-friendly.

Reply:
Warning sounds completely fair - will do!

Reply:
I disagree here, I feel like an error is reasonable even though the deployment would work because we are changing a setting that the user had set and not actually running the revision the user had specified.

Reply:
That was my gut feeling here. I'll revert that commit unless I'm told otherwise